src/dft.py

```diff
@@ -11,4 +12,17 @@ def in_order(t: T | None) -> Iterable[int]:
     >>> list(in_order(tree))
     [1, 2, 3, 4, 5]
     """
-    return  # FIXME
+    s = []
+    order = []
+    while(True):
+        if t:
+            s.append(t)
+            t = t.left
+        elif s:
+            t = s.pop()
+            order.append(t.val)
+            t = t.right
+        else:
+            break
```

Comment:
Looking great, but you could avoid the third branch with the `break` if you rewrote it as:

```python
    while t or s:
         if t:
             s.append(t)
             t = t.left
         else:
             t = s.pop()
             order.append(t.val)
             t = t.right
```